digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -1155,10 +1195,10 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
     val user = userProvider.getCurrentUser()
     if (!user.isServiceRoadMaintainer() && !user.isOperator)
       halt(Unauthorized("User is not authorized to alter serviceroad assets"))
-      val (id, pointInfo) = service.getLinearMiddlePointById(typeId, params("id").toLong)
+      val (id, pointInfo, source) = service.getLinearMiddlePointAndSourceById(typeId, params("id").toLong)

```

Comment:
Maybe this code should be in one function to be used by all the other methods 

` val (id, pointInfo, source) = service.getLinearMiddlePointAndSourceById(typeId, params("id").toLong)
      pointInfo.map {
        case middlePoint => Map("success"->true, "id" -> id, "middlePoint" -> middlePoint, "source" -> source)
      }.getOrElse(Map("success" ->false, "Reason"->"Id not found or invalid input"))`